packages/openapi-generator/src/model.ts

```diff
@@ -0,0 +1,96 @@
+import { kebabCase, last, pascalCase, unique } from '@sap-cloud-sdk/util';
+import { OpenAPIV3 } from 'openapi-types';
+import {
+  OpenApiAllOfSchema,
+  OpenApiAnyOfSchema,
+  OpenApiArraySchema,
+  OpenApiEnumSchema,
+  OpenApiNotSchema,
+  OpenApiObjectSchema,
+  OpenApiOneOfSchema,
+  OpenApiSchema
+} from './openapi-types';
+
+export function collectRefs(schema: OpenApiSchema | undefined): string[] {
+  if (!schema) {
+    return [];
+  }
+  if (isReferenceObject(schema)) {
+    return [schema.$ref];
+  }
+  return Object.values(schema)
+    .filter(value => typeof value === 'object')
+    .reduce((refs, value) => unique([...refs, ...collectRefs(value)]), []);
+}
+
+export function hasNotSchema(schema: OpenApiSchema): boolean {
```

Comment:
Two questions:
- What happens if in a `schema` an other schema is referenced? I think this is possible and we than have to follow the reference and keep on checking if there is the not property.
- If the reference problem is real, we need to consider cyclic cases.

Reply:
see related comment in the schema.

Reply:
I think cyclic references should not be an issue, because we never resolve the final schemas (as in components/schema). When there is a schema we just parse the reference string.
The `hasNotSchema` function recursively calls itself for all values in an object, so that other referenced schemas are covered.

Reply:
Clarified, fixed through type imports